polars/polars-core/src/frame/row/transpose.rs

```diff
@@ -1,25 +1,44 @@
+use std::borrow::Cow;
+
+use either::Either;
+
 use super::*;
 
 impl DataFrame {
-    pub(crate) fn transpose_from_dtype(&self, dtype: &DataType) -> PolarsResult<DataFrame> {
+    pub(crate) fn transpose_from_dtype(
+        &self,
+        dtype: &DataType,
+        keep_names_as: Option<&str>,
+        names_t: &[String],
```

Comment:
Can you rename this argument? What is meant by `t`?

Reply:
As far as I know it's a universal symbol for transpose and seemed reasonable here. I thought it would be super clear and useful in this context since at times we refer to the names of the original dataframe and at other times the names of the new transposed one.

Reply:
Let's name it `names_transposed`, then it is clear for everyone. I assume it are the names after transposing?

Reply:
We had this discussion before, I don't want single letter identifier in non-local code.

I am trying to have the code in a consistent style and having discussions on this isn't worth both our time I think.

Reply:
a. I know, I changed it!
b. This function is a helper used only in this file. How non-local are we talking?
c. Is the policy that no names can contain single-letter sub-tokens? Even where it's clear from context like `n_something` or `hstack`?

No need to answer b and c, just something to think about for the guide. Like I've said before, I'm not trying to give you a hard time, just trying to help test the policies to lead to maximal clarity. And I'm happy to help writing up guidelines if you want.

Reply:
The reason I asked initially was because when I read `names_t`, I found myself thinking; "names for what?" -> "t" -> "transpose" or "transposed"?

Do you mean the names of the columns you want to transpose or rename them after transposing. All this mental arithmetic can be prevented with slightly more explicit names. This helps many readers.

So I think `names_out` is a good parameter name now. :+1: 

Regarding c. `n_foo` is very clear when we talk about the amount of something. It is not a verb that can be in present or past tense. I think that verbs never should be changed into a single letter.  We have a `hstack` for legacy reasons (we should at least name it `h_stack`), but on the python side we all renamed `sum` to `sum_horizontal`, etc. Now, for non-local code I think it is fine to use `hstack`, `n`, `i` etc. 

I want things to be more clear when we get into function arguments and even more when we get in public domain. This isn't math, so it will not always be clear.

Reply:
> The reason I asked initially was because when I read `names_t`, I found myself thinking; "names for what?" -> "t" -> "transpose" or "transposed"?
> 
> Do you mean the names of the columns you want to transpose or rename them after transposing. All this mental arithmetic can be prevented with slightly more explicit names. This helps many readers.

Well, in this case, I think even writing out "transpose{d}" would have been wrong, since the names aren't a transpose of anything. It's the names _of_ a thing that's transposed.

> Regarding c. `n_foo` is very clear when we talk about the amount of something. It is not a verb that can be in present or past tense. I think that verbs never should be changed into a single letter. We have a `hstack` for legacy reasons (we should at least name it `h_stack`), but on the python side we all renamed `sum` to `sum_horizontal`, etc. Now, for non-local code I think it is fine to use `hstack`, `n`, `i` etc.

This may be a topic for another time, but "horizontal", "vertical" and "diagonal" are terms I'd rather nip in the bud. It's not that they're long, though I'm not a fan either, but I find myself constantly confused by them. We have perfectly good, short terms like "row", "col", "by_row", "by_col", "row_wise", "col_wise" etc that are both shorter and more natural for the domain.

> I want things to be more clear when we get into function arguments and even more when we get in public domain. This isn't math, so it will not always be clear.

Have you _seen_ the variation and inconsistency in mathematical notation? Anyway, clarity is highly subjective, so the question is always "clear to whom" and "how do you make sure"? But you know what they say, the two hardest problems in computer science are cache invalidation, naming things, and off by one errors.